src/openms/source/ANALYSIS/ID/AScore.cpp

```diff
@@ -60,191 +59,106 @@ namespace OpenMS
 
   PeptideHit AScore::compute(PeptideHit & hit, RichPeakSpectrum & real_spectrum, double fragment_mass_tolerance, bool fragment_mass_unit_ppm) const
   {
-    String with_phospho_string = hit.getSequence().toString();
-
-    // count number of phosphorylation events
-    Size number_of_phosphorylation_events = 0;
-    for (Size i = with_phospho_string.find("Phospho"); i != std::string::npos; i = with_phospho_string.find("Phospho", i + 7))
+    PeptideHit phospho = hit;
+    
+    //reset phospho
+    phospho.setScore(0);
+    
+    //String with_phospho_string = phospho.getSequence().toString();
+    String sequence_str = phospho.getSequence().toString();
+    
+    Size number_of_phosphorylation_events = numberOfPhosphoEvents_(sequence_str);
+    AASequence seq_without_phospho = removePhosphositesFromSequence_(sequence_str);
+  
+    // determine all phospho sites
+    vector<Size> sites(getSites_(seq_without_phospho));
+    Size number_of_STY = sites.size();
+    
+    if (number_of_STY < number_of_phosphorylation_events)
```

Comment:
This case shouldn't happen. This check was already there before I made the changes. I will remove that code lines.